src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableInterlocked.cs

```diff
@@ -459,8 +459,8 @@ public static bool InterlockedInitialize<T>(ref ImmutableArray<T> location, Immu
         /// <returns><c>true</c> if the key and comparison value were present in the dictionary and the update was made; <c>false</c> otherwise.</returns>
         public static bool TryUpdate<TKey, TValue>(ref ImmutableDictionary<TKey, TValue> location, TKey key, TValue newValue, TValue comparisonValue) where TKey : notnull
         {
-            var valueComparer = EqualityComparer<TValue>.Default;
             var priorCollection = Volatile.Read(ref location);
+            var valueComparer = priorCollection.ValueComparer;
```

Comment:
📝 Given we are no longer using value comparers as part of the new change, perhaps we submit this as a separate PR?

Reply:
> Given we are no longer using value comparers as part of the new change

What change are you referring to?

Reply:
https://github.com/dotnet/runtime/pull/46481#discussion_r566358634

Reply:
I'm suggesting we revert the changes to `TryUpdate` from this pull request.

Reply:
Ah, I missed that change. Yeah I think that's reasonable.

Reply:
Fixed, thanks.